lnwallet/chainfee/filtermanager.go

```diff
@@ -0,0 +1,285 @@
+	close(f.quit)
+	f.wg.Wait()
+}
+
+// fetchPeerFilters fetches our peers' feefilter values and adds them to our
+// historical values.
+func (f *filterManager) fetchPeerFilters() {
+	filterTicker := time.NewTicker(fetchFilterInterval)
+	defer filterTicker.Stop()
+
+	for {
+		select {
+		case <-filterTicker.C:
+			filters, err := f.fetchFunc(f.client)
+			if err != nil {
+				log.Errorf("Encountered err while fetching "+
+					"fee filters: %v", err)
+				return
+			}
+
+			f.updateMedian(filters)
+
+		case <-f.quit:
+			return
+		}
+	}
+}
+
+// fetchMedianFilter fetches the moving median of our historical feefilters.
+func (f *filterManager) fetchMedianFilter() (SatPerKWeight, error) {
+	f.valuesMtx.RLock()
+	defer f.valuesMtx.RUnlock()
+
+	if len(f.values) == 0 {
+		// Return errNoFilters so the caller knows to ignore this
+		// output and continue.
+		return 0, errNoFilters
+	}
+
+	// Fetch the moving median.
+	median := med(f.values)
+	return median, nil
+}
+
+type bitcoindPeerInfoResp struct {
+	Inbound      bool    `json:"inbound"`
+	MinFeeFilter float64 `json:"minfeefilter"`
+}
+
+func fetchBitcoindFilters(client *rpcclient.Client) ([]SatPerKWeight, error) {
+	resp, err := client.RawRequest("getpeerinfo", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var peerResps []bitcoindPeerInfoResp
+	err = json.Unmarshal(resp, &peerResps)
+	if err != nil {
+		return nil, err
+	}
+
+	// We filter for outbound peers since it is harder for an attacker to
+	// be our outbound peer and therefore harder for them to manipulate us
+	// into broadcasting high-fee or low-fee transactions.
+	var outboundPeerFilters []SatPerKWeight
+	for _, peerResp := range peerResps {
+		if peerResp.Inbound {
+			continue
+		}
+
+		// The value that bitcoind returns for the "minfeefilter" field
+		// is in fractions of a bitcoin that represents the satoshis
+		// per KvB. We need to convert this fraction to whole satoshis
+		// by multiplying with COIN. Then we need to convert the
+		// sats/KvB to sats/KW.
+		//
+		// Convert the sats/KvB from fractions of a bitcoin to whole
+		// satoshis.
+		filterKVByte := SatPerKVByte(
+			peerResp.MinFeeFilter * btcutil.SatoshiPerBitcoin,
+		)
+
+		if !checkFilterBounds(filterKVByte) {
+			continue
+		}
+
+		// Convert KvB to KW and add it to outboundPeerFilters.
+		outboundPeerFilters = append(
+			outboundPeerFilters, filterKVByte.FeePerKWeight(),
+		)
+	}
+
+	return outboundPeerFilters, nil
+}
+
+func fetchBtcdFilters(client *rpcclient.Client) ([]SatPerKWeight, error) {
+	resp, err := client.GetPeerInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	var outboundPeerFilters []SatPerKWeight
+	for _, peerResp := range resp {
+		// We filter for outbound peers since it is harder for an
+		// attacker to be our outbound peer and therefore harder for
+		// them to manipulate us into broadcasting high-fee or low-fee
+		// transactions.
+		if peerResp.Inbound {
+			continue
+		}
+
+		// The feefilter is already in units of sat/KvB.
```

Comment:
Ahh, didn't know we diverged there re displayed units.

Reply:
Yup, Core displays as a float